### PR TITLE
fix bug in ceph-zabbix and the bug log is:

### PR DESCRIPTION
**Pull request by zhujiantao**
2017-09-10 23:28:31.306937 7f21c2ada700  0 mgr[zabbix] Exception when sending: 'ZabbixSender' object has no attribute 'hostname'.

Signed-off-by: zhujt <zhujt@winhong.com>

### CODE REVIEW

File: src/pybind/mgr/zabbix/module.py [hunk 1]
```diff
@@ -16,23 +16,24 @@ def avg(data):
 
 
 class ZabbixSender(object):
-    def __init__(self, sender, host, port, log):
+    def __init__(self, sender, host, port, hostname,log):
         self.sender = sender
         self.host = host
         self.port = port
+	self.hostname = hostname
```
Comment (yaozongyou) [inline src/pybind/mgr/zabbix/module.py]: Is this indention right?

File: src/pybind/mgr/zabbix/module.py [hunk 2]
```diff
@@ -16,23 +16,24 @@ def avg(data):
 
 
 class ZabbixSender(object):
-    def __init__(self, sender, host, port, log):
+    def __init__(self, sender, host, port, hostname,log):
```
Comment (yaozongyou) [inline src/pybind/mgr/zabbix/module.py]: insert a space between arguments?

Comment (zhujiantao) [reply src/pybind/mgr/zabbix/module.py]: Thank you.It has been changed.

File: src/pybind/mgr/zabbix/module.py (hunk 1 above)
Comment (zhujiantao) [reply src/pybind/mgr/zabbix/module.py]: Yes, It is right.be attention of "+".

**Comment by ZVampirEM77** [comment]
Please add "Signed-off-by: zhujt <zhujt@winhong.com>" into your second commit to fix "Signed-off-by" error. Thank you.

File: src/pybind/mgr/zabbix/module.py [hunk 3]
```diff
@@ -16,23 +16,24 @@ def avg(data):
 
 
 class ZabbixSender(object):
-    def __init__(self, sender, host, port, log):
+    def __init__(self, sender, host, port, hostname, log):
         self.sender = sender
         self.host = host
         self.port = port
+	self.hostname = hostname
```
Comment (ZVampirEM77) [inline src/pybind/mgr/zabbix/module.py]: The indent in here is not right. So, can you fix it?

**Comment by ZVampirEM77** [comment]
There is a same issue in your commit 607dc4f4fcd1a8de05453bcae0dfe7563dcea561, please add "Signed-off-by: zhujt zhujt@winhong.com" into commit message

File: src/pybind/mgr/zabbix/module.py (hunk 3 above)
Comment (zhujiantao) [reply src/pybind/mgr/zabbix/module.py]: ok

**Comment by wido** [comment]
Thanks for the fixes! Could you make sure the only file changed is the Python file? You are also adding files from your editor.

**Comment by zhujiantao** [comment]
I‘m sorry. I'm a novice to github and I just want to update the module.py of ceph-zabbix. However, the sign-off is wrong when I commited.

**Comment by zhujiantao** [comment]
.idea is not part of files.All the commits are squashed into one.

2017-09-18 14:59 GMT+08:00 Wido den Hollander <notifications@github.com>:

**Comment by wido** [comment]
I've been reading the code again, but it works for me with the Luminous release and I don't see the bug.

<pre>
2017-09-18 14:59:33.226811 7fb7e8d60700 20 mgr[zabbix] Waking up for new iteration
2017-09-18 14:59:33.228557 7fb7e8d60700 20 mgr[zabbix] Sending data to Zabbix server localhost
2017-09-18 14:59:33.229149 7fb7e8d60700 20 mgr[zabbix] {'rd_bytes': 0, 'total_bytes': 31894474752L, 'overall_status_int': 0, 'osd_latency_apply_min': 0L, 'osd_latency_apply_max': 0L, 'num_osd_up': 3, 'osd_max_fill': 10.44032949873612, 'osd_backfillfull_ratio': 0.8999999761581421, 'osd_latency_commit_max': 0L, 'wr_bytes': 0, 'osd_latency_commit_min': 0L, 'num_mon': 3, 'osd_min_fill': 10.44032949873612, 'total_used_bytes': 3329888256L, 'wr_ops': 0, 'osd_nearfull_ratio': 0.8500000238418579, 'osd_latency_apply_avg': 0.0, 'overall_status': u'HEALTH_OK', 'num_pg': 0, 'osd_latency_commit_avg': 0.0, 'osd_avg_fill': 10.44032949873612, 'num_osd': 3, 'osd_full_ratio': 0.949999988079071, 'total_objects': 0L, 'num_pools': 0, 'num_osd_in': 3, 'num_pg_temp': 0, 'rd_ops': 0, 'total_avail_bytes': 28564586496L}
2017-09-18 14:59:33.241832 7fb7e8d60700 20 mgr[zabbix] Zabbix Sender: 
2017-09-18 14:59:33.242126 7fb7e8d60700 20 mgr[zabbix] Sleeping for 10 seconds
</pre>

What version (git commit) are you testing with?

<pre>
root@alpha:~# ceph -v
ceph version 12.2.0 (32ce2a3ae5239ee33d6150705cdb24d43bab910c) luminous (rc)
root@alpha:~#
<pre>

**Comment by jcsp** [comment]
I've also just looked at the code, and I can't see where it would possibly be throwing `'ZabbixSender' object has no attribute 'hostname'` -- there just aren't any self.hostname or similar references. @zhujiantao I wonder if you had already

**Comment by jcsp** [comment]
I'm going to close this for now - @zhujiantao please feel free to re-open if you can confirm that this issue is occuring on unmodified code

**Pull request closed without merging**